packages/addon/public/api/TBProMenu/implementation.js

```diff
@@ -0,0 +1,526 @@
+        left: 16px;
+      }
+    }
+  `;
+
+  var TBProMenu = class extends ExtensionCommon.ExtensionAPI {
+    getAPI(context) {
+      return {
+        TBProMenu: {
+          /**
+           * Initialize the TBProMenu.
+           *
+           * This adds the header button to the app menu initially in a non logged in state.
+           *
+           * @param {number} windowId - Id of the window the extension is installed in.
+           */
+          init(windowId) {
+            // Get the native window belonging to the specified windowId.
+            let { window } = context.extension.windowManager.get(windowId);
+
+            // Store the window for easy reuse when calling other methods
+            this.window = window;
+
+            // Inject the stylesheet
+            const stylesheet = this._createElement({
+              type: 'style',
+              text: styleSheetContent,
+            });
+
+            const menuItem = this._generateHeaderButton();
+            const banner = window.document.getElementById(
+              'appMenu-addon-banners'
+            );
+
+            this.menuItem = menuItem;
+            this.reset();
+
+            banner.parentNode.insertBefore(menuItem, banner.nextSibling);
+
+            window.document.querySelector('body').appendChild(stylesheet);
+
+            // Listen for click events on the header and emit the
+            // menu-item-clicked event with the header-click action.
+            menuItem.addEventListener('click', (e) => {
+              emitter.emit(
+                'menu-item-clicked',
+                e.target.ownerGlobal,
+                'header-click'
+              );
+            });
+          },
+
+          /**
+           * Update the header button in the main app menu.
+           *
+           * Note: The text/i18TextId and boldText/i18nBoldTextId options are
+           * mutually exclusive and the i18n id's will always take precedent if
+           * both are provided.
+           *
+           * @param {object} options - Options for updating the menu item.
+           * @param {string} options.boldText - The text that should be bold in the menuItem.
+           * @param {string} options.command - What should be populated in the oncommand attribute.
+           * @param {string} options.nav - The id of a submenu which should be
+           *  navigated to when the item is clicked. This option will override the command option.
+           * @param {string} options.i18nBoldTextId - The id of the i18n message to use for bold text.
+           * @param {string} options.i18nTextId - The id of the i18n message to use for the plain text.
+           * @param {string} options.text - The text that should be plain text in the menuItem.
+           *
+           * @returns {HTMLElement} - The submenu which was created.
+           */
+          updateMenuItem({
+            boldText,
+            command,
+            i18nBoldTextId,
+            i18nTextId,
```

Comment:
Let's do the localization from the WebExtension API instead of passing in the ids. I see you have both, but I don't think we need that flexibility here.

Reply:
@kewisch this is more or less all going through its just in the experiment code you can't call directly and have to use  `context.extension.localeData.localizeMessage` instead of the standardized api which can only be called from the background.

To be clear though are you saying you prefer to always pass pre-localized strings vs ids? i can easily update this however you like.

Reply:
Yes, I don't thin we should be calling _localize from within the experiment, as it would require pre-defined strings to exist. We should be passing in the pre-localized strings into the API.